EpsilonAuxBMS/Src/AuxCAN.c

```diff
@@ -0,0 +1,87 @@
+#include "cmsis_os.h"
+#include "AuxBMS.h"
+
+void sendHeartbeatTask(void const* arg)
+{
+    // For concurrency with reportAuxToCanTask()
+    osMutexId* canHandleMutex = (osMutexId*) arg;
+    // One time osDelayUntil initialization
+    uint32_t prevWakeTime = osKernelSysTick();
+
+    for (;;)
+    {
+        osDelayUntil(&prevWakeTime, AUX_HEARBEAT_FREQ);
+
+        if (osMutexWait(canHandleMutex, 0) != osOK)
+        {
+            continue;
+        }
+
+        // Toggle Red LED for every heartbeat sent
+        HAL_GPIO_TogglePin(RED_LED_GPIO_Port, RED_LED_Pin);
+
+        // Set CAN message address
+        hcan1.pTxMsg->StdId = AUX_HEARTBEAT_STDID;
+        // Always 1
+        hcan1.pTxMsg->Data[0] = 1;
+        // Send CAN message
+        HAL_CAN_Transmit_IT(&hcan1);
```

Comment:
Can you check if this function returns some kind of status? If it does, let's toggle the red pin every time it returns HAL_OK